### PR TITLE
Match email validation of auth/login to user create/update

### LINKED ISSUE #11399: Email validation on create/update user
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes my problem.

### Describe the Bug

Basicly whenever you create a user with an invalid email like below:
image
There is no verification if the emailadress is valid.
But when you fill in something like:
test.@gmail.com
It allows you to save but not to login.

So the issue is that some one can create an account with an invalid emailadress because no check happends.
But after that (s)he cannot login because then there is a check on a valid emailadress.

### To Reproduce

See above

### Errors Shown

```json
[
    {
        "message": "\"email\" must be a valid email",
        "extensions": {
            "code": "INVALID_PAYLOAD"
        }
    }
]
```

### What version of Directus are you using?

9.5

### What version of Node.js are you using?

14

### What database are you using?

Postgres 14

### What browser are you using?

Chrome

### What operating system are you using?

Linux

### How are you deploying Directus?

PM2

### PR DESCRIPTION
**Pull request by rijkvanzanten**
Fixes #11399

### CODE REVIEW

File: api/src/services/users.ts [hunk 1]
```diff
@@ -98,6 +99,18 @@ export class UsersService extends ItemsService {
...
 	}
 
+	private validateEmails(emails: string[]) {
+		const schema = Joi.string().email().required();
+
+		for (const email of emails) {
+			const { error } = schema.validate(email);
+
+			if (error) {
+				throw new FailedValidationError(joiValidationErrorItemToErrorExtensions(error.details![0]!));
```
Comment (jaads) [inline api/src/services/users.ts:109]: It seems to be something wrong with the parameter here 🤔 when I submit an invalid email address while creating a user, I now receive an error saying "Couldn't extract validation error type from Joi validation error item"

Comment (paescuj) [reply api/src/services/users.ts:109]: Yeah, it seems like `joiValidationErrorItemToErrorExtensions` is missing the `string` case:
```js
details: [
  {
    message: '"value" must be a valid email',
    path: [],
    type: 'string.email',
    context: [Object]
  }
]
```

Comment (paescuj) [reply api/src/services/users.ts:109]: I agree that `FailedValidationError` should probably fallback to another message on unhandled case in `joiValidationErrorItemToErrorExtensions`.

Comment (paescuj) [inline api/src/services/users.ts:109]:
```suggestion
				throw new FailedValidationError(joiValidationErrorItemToErrorExtensions(error.details[0]!));
```

Also could we maybe adapt `joiValidationErrorItemToErrorExtensions` to accept `ValidationErrorItem | undefined` so we don't have to use the non-null assertion here?

```suggestion
				throw new FailedValidationError(joiValidationErrorItemToErrorExtensions(error.details[0]));
```

**Comment by rijkvanzanten** [comment]
Gonna recreate this PR, something got funky in the commit history

**Pull request closed without merging**